src/LibraryBuilder.php

```diff
@@ -0,0 +1,143 @@
+<?php 
+
+namespace Scriptotek\GoogleBooks;
+
+use GuzzleHttp\Client;
+use Scriptotek\GoogleBooks\GoogleBooks;
+
+class LibraryBuilder
+{
+    private $data;
+    private $client;
+    private $chunk;
+    private $query;
+    private $limit;
+
+    /**
+     * Accepts a GoogleBooks dependency and sets it as a client property
+     *
+     * @param GoogleBooks $GoogleBooks
+     */
+    public function __construct(GoogleBooks $GoogleBooks)
+    {
+        $this->client = $GoogleBooks;
+    }
+
+    /**
+     * Limits the results. May be an integer between 1 and 40
+     *
+     * @param int $limit
+     * @return self
+     */
+    public function limit($limit)
+    {
+        $this->limit = $limit;
+        return $this;
+    }
+
+    /**
+     * Sets the query property
+     *
+     * @param string $query
+     * @return self
+     */
+    public function search($query)
+    {
+        $this->query = $query;
+        return $this;
+    }
+
+    /**
+     * Fetches results from the Google API
+     *
+     * @param string $id
+     * @return array
+     */
+    private function fetch($id = false)
+    {
+        if ($this->limit <= 40) {
+            if ($id !== false) {
+                return $this->client->getItem("volumes/$id");
+            }
+            return $this->client->raw('volumes', $this->getParams());
+        }
+        
+        return $this->fetchAll();
```

Comment:
Hm, I specify `->limit(50)`, I wouldn't expect to get *all* :)

Reply:
true. The limits above 40 actually don't work as intended here. Since the API has a 40 result limit, the trick here is to keep calling the api and moving the startindex. This works but it doesn't stop at 50 results, it keeps going and gets all the results. I'll prob open an issue and we can get to this some other way.

Reply:
Yeah. Perhaps just throw an exception if limit is set to more than 40 for now, so we don't forget?